src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -1702,96 +1827,225 @@ public static IEnumerable<object[]> IndexOf_NonGeneric_TestData()
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, new NonGenericClass1(), 0, 2, -1 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 0, 0, -1 };
+
+            // Struct
+            NonGenericStruct structObject = new NonGenericStruct();
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, new NonGenericStruct(), 0, 2, 0 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 0, 0, -1 };
+
+            // Interface
+            ClassWithNonGenericInterface1 interfaceObject = new ClassWithNonGenericInterface1();
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, new ClassWithNonGenericInterface1(), 0, 2, -1 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 0, 0, -1 };
+
+            // Object
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, null, 0, 1, -1 };
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, new EqualsOverrider { Value = 1 }, 0, 1, 0 };
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, new EqualsOverrider { Value = 2 }, 0, 1, -1 };
+            yield return new object[] { new object[1], null, 0, 1, 0 };
+            yield return new object[] { new object[2], null, 2, 0, -1 };
+            yield return new object[] { new object[2], null, 0, 0, -1 };
+        }
+
+        public static IEnumerable<object[]> IndexOf_Dummy_SZArray_TestData()
+        {
+            yield return new object[] { new string[] { "Hello", "Hello", "Goodbye", "Goodbye" }, new object(), 0, 4, -1, "dummyString" };
+            yield return new object[] { new string[] { "Hello", "Hello", "Goodbye", "Goodbye" }, null, 0, 4, -1, "dummyString" };
+        }
```

Comment:
I don't understand; why do you need "dummyString" here?  The 2nd and the 6th parameters are both typed as T; why is the second one insufficient to convey the type to xunit?  Especially since they're of different types in at least the first entry here.

I would really like to avoid having such "dummy" entries in the data (if some already slipped through in previous reviews, I would like to remove those).